apps/api/src/app/events/usecases/add-job/add-delay-job.usecase.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@nestjs/common';
+import { JobEntity, JobRepository, JobStatusEnum } from '@novu/dal';
+import { DigestUnitEnum, StepTypeEnum, DelayTypeEnum } from '@novu/shared';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { WorkflowQueueService } from '../../services/workflow.queue.service';
+import { AddJobCommand } from './add-job.command';
+import { differenceInMilliseconds } from 'date-fns';
+
+@Injectable()
+export class AddDelayJob {
+  constructor(private jobRepository: JobRepository, private workflowQueueService: WorkflowQueueService) {}
+
+  public async execute(command: AddJobCommand): Promise<boolean> {
+    const data = await this.jobRepository.findById(command.jobId);
+
+    if (!data) {
+      return false;
+    }
+
+    const isDelayStep = data.type === StepTypeEnum.DELAY;
+
+    if (!isDelayStep) {
+      return false;
+    }
+
+    await this.jobRepository.updateStatus(data._id, JobStatusEnum.DELAYED);
+
+    const delay = this.calculateDelayAmount(data);
+
+    await await this.workflowQueueService.addToQueue(data._id, data, delay);
+
+    return true;
+  }
+
+  private checkValidDelayOverride(data: JobEntity): boolean {
+    if (!data.overrides?.delay) {
+      return false;
+    }
+    const values = Object.values(DigestUnitEnum);
+
+    return (
+      typeof data.overrides.delay.amount === 'number' &&
```

Comment:
What brings the need to validate the type of the variable? Is there any chance it could be overriden for something that is not a number? That would make me think we might be missing a validation upstream. 🤔

Reply:
Good question, I would like to check if @ainouzgali knows why 😄